doc/components/fair-share.rst

```diff
@@ -0,0 +1,396 @@
+       └── leaf.3.2
+
+Let's walk through the weight calculation for one of the users in this
+hierarchy: ``leaf.3.2``:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 15 10 10
+
+   * - parent
+     - name
+     - shares
+     - usage
+   * - account3
+     - leaf.3.2
+     - 10
+     - 1
+
+We can calculate ``leaf.3.2``'s weighted shares with the equation we defined
+above:
+
+.. math::
+
+  \text{s_weighted}_{leaf.3.2} = \frac{\text{shares}_{leaf.3.2}}{\text{shares}_{leaf.3.2+siblings}} = \frac{\text{10}}{\text{110}} = \text{0.0909091}
+
+.. math::
+
+  \text{u_weighted}_{leaf.3.2} = \frac{\text{usage}_{leaf.3.2}}{\text{usage}_{leaf.3.2+siblings}} = \frac{\text{1}}{\text{1}} = \text{1}
+
+So, the final weight for ``leaf.3.2`` can be calculated as the following:
+
+.. math::
+
+  \text{weight}_{leaf.3.2} = \frac{\text{s_weighted}_{leaf.3.2}}{\text{u_weighted}_{leaf.3.2}} = \frac{\text{0.0909091}}{\text{1}} = \text{0.0909091}
+
+We apply the same process to a bank that holds users. Take ``account3`` for example:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 15 10 10
+
+   * - parent
+     - name
+     - shares
+     - usage
+   * - root
+     - account3
+     - 10
+     - 1
+
+.. math::
+
+  \text{s_weighted}_{account3} = \frac{\text{shares}_{account3}}{\text{shares}_{account3+siblings}} = \frac{\text{10}}{\text{1110}} = \text{0.0909091}
+
+.. math::
+
+  \text{u_weighted}_{account3} = \frac{\text{usage}_{account3}}{\text{usage}_{account3+siblings}} = \frac{\text{1}}{\text{133}} = \text{0.0075188}
+
+.. math::
+
+  \text{weight}_{account3} = \frac{\text{s_weighted}_{account3}}{\text{u_weighted}_{account3}} = \frac{\text{0.00900901}}{\text{0.0075188}} = \text{1.1982}
+
+We repeat this process for every user and sub-bank in the hierarchy:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 15 10
+
+   * - parent
+     - name
+     - weight
+   * - root
+     - 
+     - 
+   * - root
+     - account1
+     - 0.990246
+   * - account1
+     - leaf.1.1
+     - 0.109009
+   * - account1
+     - leaf.1.2
+     - 0.0990991
+   * - account1
+     - leaf.1.3
+     - 10.9009
+   * - root
+     - account2
+     - 1.08927
+   * - account2
+     - leaf.2.1
+     - 1.25
+   * - account2
+     - leaf.2.2
+     - 0.333333
+   * - root
+     - account3
+     - 1.1982
+   * - account3
+     - leaf.3.1
+     - 1.84467
+   * - account3
+     - leaf.3.2
+     - 0.909091
+
+After sorting the banks in descending order, we see that ``account3`` has the
+highest weight at 1.1982, followed by ``account2`` at 1.08927, and then finally
+``account1`` at 0.990246. Each bank's set of users are then sorted by their
+weight. If we look at ``account3``, we see that ``leaf.3.1`` has the highest
+weight at 1.84467. Therefore, ``leaf.3.1`` receives the highest rank of all of
+the users. ``leaf.3.2`` follows with the second highest rank, and ``account3``
+is now fully sorted. We repeat this process for the rest of the banks in the
+hierarchy:
+
+.. list-table::
```

Comment:
| rank      | parent      | name      | weight      |
| ------------- | ------------- | ------------- | ------------- |
|  | root | account3 | 1.1982 |
| 1 |  |  |  |
| 2 |  |  |  |
| 3 |  |  |  |
| 4 |  |  |  |
| 5 |  |  |  |
| 6 |  |  |  |
| 7 |  |  |  |

Reply:
Sure! I can copy this table over. That makes sense

Reply:
Just force-pushed up a change to use this table that includes the rank next to the users